[CodeLLDB] 'Not possible to change sign vimspectorNonActivePC'

### Description

A video clip: https://github.com/user-attachments/assets/3ef44181-54a2-4ed2-b4c6-ecb6c73c26ae

Sometimes, stepping to the next instruction in disassembly view fails for a few steps.
This happens for the next few steps in a row and vim's error message disappears in a couple of seconds even if I try to stay at the current insturction by not pressing any button.

This happens on `CodeLLDB`. `vscode-cpptools` works fine without any issue.

I see this symptom at specific disassembly line. It's very easy to reproduce the symptom.

### Is it reproducible in vim?

Reproducible in Vim

### Works with clean config?

Yes

### Sample project works?

Not reproducible with sample project

### Minimal Reproduction

List of steps to reproduce:

1. `git clone https://github.com/wind0204/carbtester && cd carbtester`
2. `CC=clang-18 cmake -B build-RelWithDebInfo -DCMAKE_BUILD_TYPE=RelWithDebInfo`
3. `cmake --build build-RelWithDebInfo -- -j $(nproc)`
4. `vim src/info_struct/info_struct-tree.c`
5. Go to line number `330` and place a breakpoint
6. `:call vimspector#Launch()`
7. Choose `7` ( 7: launch-tester-RelWithDebInfo-CodeLLDB )
8. Type any integer lower than 64 and greater than 0 for the CommandLineArgs
9. Open the disassembly view
10. Step to next instruction a few times and the symptom can be seen when you step to the next instruction of the instruction described below
```asm
0x000055555555ca45: 0F 87 AD 02 00 00                           ja     0x8cf8  ; <+712> at info_struct-tree.c:412:1
```

### Expected Behaviour

Step to the next instruction without failure, and without an error message shown.

### Actual Behaviour

The cursor stays at the current(previous) instruction for a few steps in a row, and a lot of error messages are printed on vim in each step.

### Additional information


`.vimspector.json` of my project:
```json
{
  "$schema": "https://puremourning.github.io/vimspector/schema/vimspector.schema.json#",
  "configurations": {
    "launch-Debug-vscode-cpptools": {
      "adapter": "vscode-cpptools",
      "variables": {
      },
      "configuration": {
        "request": "launch",
        "program": "${workspaceRoot}/build-Debug/Debug/carbonara",
        "args": [ "*${CommandLineArgs}" ],
        "stopOnEntry": true
      },
      "breakpoints": {
        "exception": {
          "caught": "",
          "uncaught": "Y"
        }
      }
    },
    "launch-RelWithDebInfo-vscode-cpptools": {
      "adapter": "vscode-cpptools",
      "variables": {
      },
      "configuration": {
        "request": "launch",
        "program": "${workspaceRoot}/build-RelWithDebInfo/RelWithDebInfo/carbonara",
        "args": [ "*${CommandLineArgs}" ]
      },
      "breakpoints": {
        "exception": {
          "caught": "N",
          "uncaught": "Y"
        }
      }
    },
    "launch-tester-Debug-vscode-cpptools": {
      "adapter": "vscode-cpptools",
      "variables": {
      },
      "configuration": {
        "request": "launch",
        "program": "${workspaceRoot}/build-Debug/Debug/tester",
        "args": [ "*${CommandLineArgs}" ],
        "stopOnEntry": true
      },
      "breakpoints": {
        "exception": {
          "caught": "",
          "uncaught": "Y"
        }
      }
    },
    "launch-tester-RelWithDebInfo-vscode-cpptools": {
      "adapter": "vscode-cpptools",
      "variables": {
      },
      "configuration": {
        "request": "launch",
        "program": "${workspaceRoot}/build-RelWithDebInfo/RelWithDebInfo/tester",
        "args": [ "*${CommandLineArgs}" ],
        "stopOnEntry": true
      },
      "breakpoints": {
        "exception": {
          "caught": "",
          "uncaught": "Y"
        }
      }
    },
    "launch-Debug-CodeLLDB": {
      "adapter": "CodeLLDB",
      "variables": {
      },
      "configuration": {
        "request": "launch",
        "program": "${workspaceRoot}/build-Debug/Debug/carbonara",
        "args": [ "*${CommandLineArgs}" ],
        "stopOnEntry": true
      },
      "breakpoints": {
        "exception": {
          "caught": "",
          "uncaught": "Y"
        }
      }
    },
    "launch-RelWithDebInfo-CodeLLDB": {
      "adapter": "CodeLLDB",
      "variables": {
      },
      "configuration": {
        "request": "launch",
        "program": "${workspaceRoot}/build-RelWithDebInfo/RelWithDebInfo/carbonara",
        "args": [ "*${CommandLineArgs}" ]
      },
      "breakpoints": {
        "exception": {
          "caught": "N",
          "uncaught": "Y"
        }
      }
    },
    "launch-tester-Debug-CodeLLDB": {
      "adapter": "CodeLLDB",
      "variables": {
      },
      "configuration": {
        "request": "launch",
        "program": "${workspaceRoot}/build-Debug/Debug/tester",
        "args": [ "*${CommandLineArgs}" ],
        "stopOnEntry": true
      },
      "breakpoints": {
        "exception": {
          "caught": "",
          "uncaught": "Y"
        }
      }
    },
    "launch-tester-RelWithDebInfo-CodeLLDB": {
      "adapter": "CodeLLDB",
      "variables": {
      },
      "configuration": {
        "request": "launch",
        "program": "${workspaceRoot}/build-RelWithDebInfo/RelWithDebInfo/tester",
        "args": [ "*${CommandLineArgs}" ],
        "stopOnEntry": true
      },
      "breakpoints": {
        "exception": {
          "caught": "",
          "uncaught": "Y"
        }
      }
    }
  }
}
```


### Installation Type

Vim packages (git clone)

### Vimspector version

f842ce1b44af1fa7e40f05fc2cedecaefcf4f852

### Debug Info

```Text
Vimspector Debug Info
--------------------------------------------------------------------------------
ConnectionType: job
Adapter: 
--------------------------------------------------------------------------------
{
  "command": [
    "/home/dewr/.vim/pack/my_favorites/opt/vimspector/gadgets/linux/CodeLLDB/adapter/codelldb"
  ],
  "configuration": {
    "args": [],
    "cargo": {},
    "cwd": "/home/dewr/Products/carbonara",
    "env": {},
    "name": "lldb",
    "terminal": "integrated",
    "type": "lldb"
  },
  "name": "CodeLLDB",
  "type": "CodeLLDB",
  "env": {},
  "cwd": "/home/dewr/Products/carbonara"
}
--------------------------------------------------------------------------------
Configuration: 
--------------------------------------------------------------------------------
{
  "adapter": "CodeLLDB",
  "configuration": {
    "request": "launch",
    "program": "/home/dewr/Products/carbonara/build-RelWithDebInfo/RelWithDebInfo/tester",
    "args": [
      "36"
    ],
    "stopOnEntry": true
  },
  "breakpoints": {
    "exception": {
      "caught": "",
      "uncaught": "Y"
    }
  }
}
--------------------------------------------------------------------------------
API Prefix: 
Launch/Init: True / True
Workspace Root: /home/dewr/Products/carbonara
Launch Config: 
--------------------------------------------------------------------------------
{
  "args": [
    "36"
  ],
  "cargo": {},
  "cwd": "/home/dewr/Products/carbonara",
  "env": {},
  "name": "lldb",
  "terminal": "integrated",
  "type": "lldb",
  "request": "launch",
  "program": "/home/dewr/Products/carbonara/build-RelWithDebInfo/RelWithDebInfo/tester",
  "stopOnEntry": true
}
--------------------------------------------------------------------------------
Server Capabilities: 
--------------------------------------------------------------------------------
{
  "exceptionBreakpointFilters": [
    {
      "default": true,
      "filter": "cpp_throw",
      "label": "C++: on throw",
      "supportsCondition": true
    },
    {
      "default": false,
      "filter": "cpp_catch",
      "label": "C++: on catch",
      "supportsCondition": true
    }
  ],
  "supportTerminateDebuggee": true,
  "supportsCancelRequest": true,
  "supportsCompletionsRequest": true,
  "supportsConditionalBreakpoints": true,
  "supportsConfigurationDoneRequest": true,
  "supportsDataBreakpoints": true,
  "supportsDelayedStackTraceLoading": true,
  "supportsDisassembleRequest": true,
  "supportsEvaluateForHovers": true,
  "supportsExceptionFilterOptions": true,
  "supportsExceptionInfoRequest": true,
  "supportsFunctionBreakpoints": true,
  "supportsGotoTargetsRequest": true,
  "supportsHitConditionalBreakpoints": true,
  "supportsInstructionBreakpoints": true,
  "supportsLogPoints": true,
  "supportsReadMemoryRequest": true,
  "supportsSetVariable": true,
  "supportsSteppingGranularity": true,
  "supportsWriteMemoryRequest": true
}
--------------------------------------------------------------------------------
Line Breakpoints: 
--------------------------------------------------------------------------------
{
  "/home/dewr/Products/carbonara/src/info_struct/info_struct-tree.c": [
    {
      "state": "ENABLED",
      "line": 330,
      "options": {},
      "is_instruction_breakpoint": null,
      "sign_id": 1,
      "server_bp": {
        "0": {
          "id": 1,
          "line": 331,
          "message": "Resolved locations: 1",
          "verified": true
        }
      }
    }
  ],
  "_vimspector_tmp/0/@_start[0]": []
}
--------------------------------------------------------------------------------
Func Breakpoints: 
--------------------------------------------------------------------------------
[]
--------------------------------------------------------------------------------
Ex Breakpoints: 
--------------------------------------------------------------------------------
{
  "filters": [
    "cpp_throw"
  ]
}
--------------------------------------------------------------------------------
```

### Log file

```Text
[vimspector.log](https://github.com/user-attachments/files/22558452/vimspector.log)
```

### Vim version

```Text
VIM - Vi IMproved 9.1 (2024 Jan 02, compiled Sep 05 2025 19:44:46)
Included patches: 1-16, 647, 678, 697
Modified by team+vim@tracker.debian.org
Compiled by team+vim@tracker.debian.org
Huge version without GUI.  Features included (+) or not (-):
+acl               +file_in_path      +mouse_urxvt       -tag_any_white
+arabic            +find_in_path      +mouse_xterm       -tcl
+autocmd           +float             +multi_byte        +termguicolors
+autochdir         +folding           +multi_lang        +terminal
-autoservername    -footer            -mzscheme          +terminfo
-balloon_eval      +fork()            +netbeans_intg     +termresponse
+balloon_eval_term +gettext           +num64             +textobjects
-browse            -hangul_input      +packages          +textprop
++builtin_terms    +iconv             +path_extra        +timers
+byte_offset       +insert_expand     -perl              +title
+channel           +ipv6              +persistent_undo   -toolbar
+cindent           +job               +popupwin          +user_commands
-clientserver      +jumplist          +postscript        +vartabs
-clipboard         +keymap            +printer           +vertsplit
+cmdline_compl     +lambda            +profile           +vim9script
+cmdline_hist      +langmap           -python            +viminfo
+cmdline_info      +libcall           +python3           +virtualedit
+comments          +linebreak         +quickfix          +visual
+conceal           +lispindent        +reltime           +visualextra
+cryptv            +listcmds          +rightleft         +vreplace
+cscope            +localmap          -ruby              +wildignore
+cursorbind        -lua               +scrollbind        +wildmenu
+cursorshape       +menu              +signs             +windows
+dialog_con        +mksession         +smartindent       +writebackup
+diff              +modify_fname      +sodium            -X11
+digraphs          +mouse             -sound             +xattr
-dnd               -mouseshape        +spell             -xfontset
-ebcdic            +mouse_dec         +startuptime       -xim
+emacs_tags        +mouse_gpm         +statusline        -xpm
+eval              -mouse_jsbterm     -sun_workshop      -xsmp
+ex_extra          +mouse_netterm     +syntax            -xterm_clipboard
+extra_search      +mouse_sgr         +tag_binary        -xterm_save
-farsi             -mouse_sysmouse    -tag_old_static    
   system vimrc file: "/etc/vim/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
  fall-back for $VIM: "/usr/share/vim"
Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H -Wdate-time -g -O2 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -ffile-prefix-map=/build/vim-xEtJBd/vim-9.1.0016=. -flto=auto -ffat-lto-objects -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection -fdebug-prefix-map=/build/vim-xEtJBd/vim-9.1.0016=/usr/src/vim-2:9.1.0016-1ubuntu7.9 -DSYS_VIMRC_FILE=\"/etc/vim/vimrc\" -DSYS_GVIMRC_FILE=\"/etc/vim/gvimrc\" -D_REENTRANT -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 
Linking: gcc -Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -Wl,-z,relro -Wl,-z,now -Wl,--as-needed -o vim -lm -ltinfo -lselinux -lsodium -lacl -lattr -lgpm -L/usr/lib/python3.12/config-3.12-x86_64-linux-gnu -lpython3.12 -ldl -lm
```

### Python version

_No response_

### Neovim diagnostics

```Text

```

### Operating System

Ubuntu 24.04.3 LTS ( Linux ga-z87-hd3 6.8.0-84-generic #84-Ubuntu SMP PREEMPT_DYNAMIC Fri Sep  5 22:36:38 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux )

### Declaration

- [x] I have read and understood [CONTRIBUTING.md](https://github.com/puremourning/vimspector/blob/master/CONTRIBUTING.md#diagnostics)
- [x] I have read and understood the [CODE_OF_CONDUCT.md](https://github.com/puremourning/vimspector/blob/master/CODE_OF_CONDUCT.md)

I tried your repro steps, but

```
ben@BenMBP2021 carbtester % ls
cmake                    COPYING                  README                   wayland-protocols
CMakeLists.txt           extern                   src                      WaylandProtocolGen.cmake
ben@BenMBP2021 carbtester % cmake -B build -S .
-- The C compiler identification is AppleClang 17.0.0.17000319
-- The CXX compiler identification is AppleClang 17.0.0.17000319
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- CMAKE_MODULE_PATH: /Users/ben/Development/vimspector/carbtester/cmake/find-modules;/usr/share/ECM/find-modules
CMake Error at CMakeLists.txt:66 (message):
  Please provide the environment variable VULKAN_SDK


-- Configuring incomplete, errors occurred!
ben@BenMBP2021 carbtester %```

--- a/python3/vimspector/stack_trace.py
+++ b/python3/vimspector/stack_trace.py
@@ -706,7 +706,8 @@ class StackTraceView( object ):
                            self._buf.name,
                            line )
 
-          if utils.BufferExists( frame[ 'source' ][ 'path' ] ):
+          if ( utils.BufferExists( frame[ 'source' ][ 'path' ] )
+               and frame[ 'line' ] ):
             sign_id = len( self._top_of_stack_signs ) + 100
             self._top_of_stack_signs.append( sign_id )
             signs.PlaceSign( sign_id,

--- a/support/test/cpp/simple_c_program/.vimspector.json
+++ b/support/test/cpp/simple_c_program/.vimspector.json
@@ -51,6 +51,11 @@
     },
     "CodeLLDB": {
       "adapter": "CodeLLDB",
+      "variables": {
+        "BUILDME": {
+          "shell": "g++ -o ${workspaceRoot}/test -g -std=c++17 ${workspaceRoot}/test_c.cpp"
+        }
+      },
       "configuration": {
         "request": "launch",
         "expressions": "native",

